File: recipe_archive/android/app/src/main/kotlin/com/recipeArchive/recipe_archive/AppLogger.kt
package com.recipeArchive.recipe_archive

import android.util.Log

/**
 * Centralized structured logging for RecipeArchive Android app
 * Provides consistent logging categories and methods across the application
 */
object AppLogger {

    private const val TAG_PREFIX = "RecipeArchive"

    /**
     * Logger for Share Extension operations
     */
    object ShareExtension {
        private const val TAG = "$TAG_PREFIX:ShareExtension"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for WebView content loading
     */
    object WebView {
        private const val TAG = "$TAG_PREFIX:WebView"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for network operations (S3, API calls)
     */
    object Network {
        private const val TAG = "$TAG_PREFIX:Network"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for recipe parsing operations
     */
    object Parser {
        private const val TAG = "$TAG_PREFIX:Parser"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for local storage operations
     */
    object Storage {
        private const val TAG = "$TAG_PREFIX:Storage"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for authentication operations
     */
    object Auth {
        private const val TAG = "$TAG_PREFIX:Auth"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }

    /**
     * Logger for Flutter MethodChannel bridge
     */
    object Flutter {
        private const val TAG = "$TAG_PREFIX:Flutter"

        fun debug(message: String) = Log.d(TAG, message)
        fun info(message: String) = Log.i(TAG, message)
        fun warning(message: String) = Log.w(TAG, message)
        fun error(message: String, throwable: Throwable? = null) {
            if (throwable != null) {
                Log.e(TAG, message, throwable)
            } else {
                Log.e(TAG, message)
            }
        }
    }
}

File: recipe_archive/ios/Shared/AppLogger.swift
//
//  AppLogger.swift
//  RecipeArchive
//
//  Centralized structured logging using Apple's os.Logger
//

import Foundation
import os

/// Centralized logger factory for RecipeArchive app
/// Uses Apple's native os.Logger for structured logging with privacy controls
public enum AppLogger {

    /// Logger for Share Extension operations
    public static let shareExtension = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "ShareExtension"
    )

    /// Logger for WKWebView content loading
    public static let webView = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "WebView"
    )

    /// Logger for network operations (S3, API calls)
    public static let network = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "Network"
    )

    /// Logger for recipe parsing operations
    public static let parser = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "Parser"
    )

    /// Logger for local storage operations
    public static let storage = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "Storage"
    )

    /// Logger for authentication operations
    public static let auth = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "Auth"
    )

    /// Logger for Flutter MethodChannel bridge
    public static let flutter = Logger(
        subsystem: Bundle.main.bundleIdentifier ?? "com.RecipeArchive",
        category: "Flutter"
    )
}
